torch_geometric/nn/conv/gin_conv.py

```diff
@@ -141,10 +142,15 @@ def __init__(self, nn: Callable, eps: float = 0., train_eps: bool = False,
         else:
             self.register_buffer('eps', torch.Tensor([eps]))
         if edge_dim is not None:
-            if hasattr(self.nn[0], 'in_features'):
-                in_channels = self.nn[0].in_features
+            if (isinstance(self.nn, torch.nn.Sequential)
```

Comment:
I don't actually like this logic much, I think it might be nicer to just require the input channels to be provided. But it at least solves the bug.

Reply:
LazyLinear does not work since it can only infer the input channels, not the output channels.

Reply:
Ah, good point